src/wp-includes/blocks.php

```diff
@@ -760,6 +760,108 @@ function get_hooked_blocks( $name ) {
 	return $hooked_blocks;
 }
 
+function insert_hooked_blocks( $block ) {
+	$hooked_blocks = get_hooked_blocks( $block['blockName'] );
+	foreach ( $hooked_blocks as $hooked_block_type => $relative_position ) {
+		$hooked_block = array(
+			'blockName'    => $hooked_block_type,
+			'attrs'        => array(),
+			'innerHTML'    => '',
+			'innerContent' => array(),
+			'innerBlocks'  => array(),
+		);
+		// Need to pass full current block, (possibly its parent), relative position, and hooked_block.
+		$block = insert_hooked_block( $hooked_block, $relative_position, $block );
+	}
+	return $block;
+}
+
+/**
+ * Auto-inserts a block next to a given "anchor" block.
+ *
+ * This is a helper function used in the implementation of block hooks.
+ * It is not meant for public use.
+ *
+ * The auto-inserted block can be inserted before or after the anchor block,
+ * or as the first or last child of the anchor block.
+ *
+ * Note that the this mutates the automatically inserted block's
+ * designated parent block by inserting into the parent's `innerBlocks` array,
+ * and by updating the parent's `innerContent` array accordingly.
+ *
+ * @since 6.4.0
```

Comment:
It's possible to mark it as private which will exclude it from the documentation:
```suggestion
 * @since 6.4.0
 * @access private
```

Reply:
I was actually considering keeping it public 🤔 It's not like I expect people to use it a lot, but it's going to be part of a few call stacks (hooked block insertion for both templates and patterns) so I thought it makes sense to have it documented.

Reply:
Works for me. In that case the following line should go away:

>  It is not meant for public use.

Reply:
Ah, good catch! That's a leftover from GB times.

I should probably re-read the PHPDoc at some point 😬

Reply:
https://github.com/WordPress/wordpress-develop/pull/5240/commits/d1cf61ed6cbff74eaee4766934ade067d9177d53